packages/extensionmanager/src/model.ts

```diff
@@ -708,8 +807,16 @@ export class ListModel extends VDomModel {
   private _searchResult: IEntry[];
   private _pendingActions: Promise<any>[] = [];
   private _debouncedUpdate: Debouncer<void, void>;
+
+  private _listMode: 'black' | 'white' | null;
+  private _blacklistMap: Map<string, IListEntry>;
```

Comment:
Curious why this is not just an `Array<IListEntry>`. The string key just seems to be the name?

Reply:
I was showing before more details in the UI (reasons for blaklistings...). Then we decided to not show that. I propose to leave it as such, I like knowing that it is an array of listentry, not simply an array of string.

Reply:
Yeah, so it could be an `Array<IListEntry>` not an `Array<string>`? I think you have this as an existing todo anyway: https://github.com/jupyterlab/jupyterlab/pull/7989/files/bc7b5d1efe06ae2c2feb5161b56d3f00ce30f70a#diff-4c52b79547ae3ec18aa8f0172bca4f58R542

Reply:
yep. my first implementation was not using regexp so a Map was better. Now with the regexp an Array is enough. I will pass on all current and new TODO tomorrow.

Reply:
Replacing `tomorrow` with `today` (always better :)) - Map have been replace with Array. Resolving.